.github/actions/setup-env/action.yml

```
@@ -14,33 +27,52 @@ outputs:
 runs:
   using: composite
   steps:
-    - name: Set up pdm with python ${{ inputs.python-version }}
+    - name: Download uv lock
+      uses: actions/cache/restore@v4
+      if: ${{ inputs.use-cached-uv-lock != 'false' }}
+      id: restore-uv-lock
+      with:
+        key: uv-lock-resolved-${{ inputs.uv-resolution-strategy }}
+        path: uv.lock
+        restore-keys: |
+          uv-lock-resolved-${{ inputs.uv-resolution-strategy }}
+        enableCrossOsArchive: true
+        fail-on-cache-miss: ${{ inputs.update-uv-lock != 'true' }}
+
+    - name: Set up uv with python ${{ inputs.python-version }}
       id: setup-python
-      uses: pdm-project/setup-pdm@v4
+      uses: astral-sh/setup-uv@v6
       with:
-        allow-python-prereleases: true
+        version: '0.8.14'
+        enable-cache: true
+        # this doesn't install python but pins the uv version; its the same as providing UV_PYTHON
         python-version: ${{ inputs.python-version }}
-        version: "2.20.1"  # last version to support python 3.8
-        cache: false
 
-    - name: Disable PDM version check
+    - name: Install python ${{ inputs.python-version }}
       shell: bash
+      env:
+        UV_PYTHON_DOWNLOADS: automatic
       run: |
-        pdm config check_update false
+        uv python install ${{ inputs.python-version }}
 
-    - name: Ignore saved pythons
+    - name: Install nox
       shell: bash
       run: |
-        echo "PDM_IGNORE_SAVED_PYTHON=1" >> $GITHUB_ENV
+        uv tool install nox
 
-    - name: Install nox
+    - name: Lock dependencies
+      id: lock-uv-deps
+      if: ${{ inputs.update-uv-lock }}
       shell: bash
-      run: pip install nox
+      env:
+        UV_LOCKED: 0
+      run: uv lock --refresh
 
     - name: Set python version
       id: python-version
       shell: bash
-      run: echo "python-version=$(python -c 'import sys; print(".".join(map(str,sys.version_info[:2])))')" >> $GITHUB_OUTPUT
-
-    - name: Configure cache
-      uses: ./.github/actions/cache-pdm
+      env:
+        UV_VENV_CLEAR: 1
+      run: |
+        uv venv .venv
+        echo "python-version=$($(uv python find) -c 'import sys; print(".".join(map(str,sys.version_info[:2])))')" >> $GITHUB_OUTPUT
```

```suggestion
        echo "python-version=$(uv run python -c 'import sys; print(".".join(map(str,sys.version_info[:2])))')" >> $GITHUB_OUTPUT
```
Not sure if there are subtle differences, but I'd *assume* it's the same?

---

https://github.com/DisnakeDev/disnake/pull/1362/commits/ca2068934c451d2ffd8e1a1d1e01724a547ee3c3